backend/tracim_backend/applications/share/controller.py

```diff
@@ -0,0 +1,215 @@
+    def disable_content_share(self, context, request: TracimRequest, hapic_data=None) -> None:
+        """
+        remove a file share
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = ShareLib(
+            current_user=request.current_user, session=request.dbsession, config=app_config
+        )
+        api.disable_content_share(request.current_content, hapic_data.path.share_id)
+        return
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__CONTENT_FILE_ENDPOINTS])
+    @hapic.input_path(ShareTokenPathSchema())
+    @hapic.output_body(ContentShareInfoSchema())
+    def guest_download_info(
+        self, context, request: TracimRequest, hapic_data=None
+    ) -> ContentShareInContext:
+        """
+        get content file info
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = ShareLib(current_user=None, session=request.dbsession, config=app_config)
+        content_share = api.get_content_share_by_token(
+            share_token=hapic_data.path.share_token
+        )  # type: ContentShare
+
+        # TODO - G.M - 2019-08-01 - verify in access to content share can be granted
+        # we should considered do these check at decorator level
+        if content_share.content.type not in shareables_content_type:
+            raise ContentTypeNotAllowed()
+
+        return api.get_content_share_in_context(content_share)
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__CONTENT_FILE_ENDPOINTS])
+    @hapic.handle_exception(WrongSharePassword, HTTPStatus.FORBIDDEN)
+    @hapic.input_path(ShareTokenWithFilenamePathSchema())
+    @hapic.input_headers(TracimSharePasswordHeaderSchema())
+    @hapic.output_file([])
+    def guest_download_file(self, context, request: TracimRequest, hapic_data=None) -> HapicFile:
+        """
+        remove a file share
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = ShareLib(current_user=None, session=request.dbsession, config=app_config)
+        content_share = api.get_content_share_by_token(
+            share_token=hapic_data.path.share_token
+        )  # type: ContentShare
+
+        # TODO - G.M - 2019-08-01 - verify in access to content share can be granted
+        # we should considered do these check at decorator level
+        api.check_password(content_share, password=hapic_data.headers.tracim_share_password)
+        if content_share.content.type not in shareables_content_type:
+            raise ContentTypeNotAllowed()
+
+        try:
+            file = DepotManager.get().get(content_share.content.depot_file)
+        except IOError as exc:
+            raise TracimFileNotFound(
+                "file related to revision {} of content {} not found in depot.".format(
+                    content_share.content.revision_id, content_share.content.content_id
+                )
+            ) from exc
```

Comment:
Why do you manage case where file is not present on disk ? If we really want to mange this case, it is a global case and code should be moved otherwise - i think.

Reply:
We should not allow to run code which will run template which just does not exist.
We do the same check for other template too in config.

Reply:
I don't understand why you're speaking about "template". This code deal with content's file, not template. Or i miss something ...

About my original comment: What i'm saying is: A controller views should not have to manage the case "depot file cannot be retrieved because there is corrupted data". Only `@hapic.handle_exception(ParentErrorOfDepotProblems)` is enough.

Reply:
Oh, sorry. it remind me another piece of code.
There is currently not such thing about depot in ContentLib, not sure how we should handle this.

Reply:
I'm considering this controller view should not manage this case because the case where binary file is not on disk should not happen because:

* When File `Content` type  do not own file: we should not be able to share it
* When File `Content` type has been deleted: shares are they deleted ? They must.

But, as you say the case "binary file can be absent on disk when deal with DepotManager" is a general case on Tracim. So i suggest opening a technical debt issue to think about all of that.

Reply:
ticket about moving this code here : https://github.com/tracim/tracim/issues/2168